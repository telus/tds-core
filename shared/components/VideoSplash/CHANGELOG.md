# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [1.1.32](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.1.31...@tds/shared-video-splash@1.1.32) (2020-07-22)

**Note:** Version bump only for package @tds/shared-video-splash





## [1.1.31](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.1.30...@tds/shared-video-splash@1.1.31) (2020-04-14)

**Note:** Version bump only for package @tds/shared-video-splash





## [1.1.30](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.1.29...@tds/shared-video-splash@1.1.30) (2020-02-24)

**Note:** Version bump only for package @tds/shared-video-splash





## [1.1.29](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.1.28...@tds/shared-video-splash@1.1.29) (2020-02-03)

**Note:** Version bump only for package @tds/shared-video-splash





## [1.1.28](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.1.27...@tds/shared-video-splash@1.1.28) (2020-01-27)

**Note:** Version bump only for package @tds/shared-video-splash





## [1.1.27](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.1.26...@tds/shared-video-splash@1.1.27) (2020-01-23)

**Note:** Version bump only for package @tds/shared-video-splash





## [1.1.26](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.1.25...@tds/shared-video-splash@1.1.26) (2020-01-22)

**Note:** Version bump only for package @tds/shared-video-splash





## [1.1.25](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.1.24...@tds/shared-video-splash@1.1.25) (2020-01-17)

**Note:** Version bump only for package @tds/shared-video-splash





## [1.1.24](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.1.23...@tds/shared-video-splash@1.1.24) (2020-01-07)


### Bug Fixes

* **shared-video-splash:** enable focus outline on play button ([3365859](https://github.com/telus/tds-core/commit/3365859))





## [1.1.23](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.1.22...@tds/shared-video-splash@1.1.23) (2019-11-20)

**Note:** Version bump only for package @tds/shared-video-splash





## [1.1.22](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.1.21...@tds/shared-video-splash@1.1.22) (2019-11-06)

**Note:** Version bump only for package @tds/shared-video-splash





## [1.1.21](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.1.20...@tds/shared-video-splash@1.1.21) (2019-10-18)

**Note:** Version bump only for package @tds/shared-video-splash





## [1.1.20](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.1.19...@tds/shared-video-splash@1.1.20) (2019-10-11)

**Note:** Version bump only for package @tds/shared-video-splash





## [1.1.19](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.1.18...@tds/shared-video-splash@1.1.19) (2019-10-09)

**Note:** Version bump only for package @tds/shared-video-splash





## [1.1.18](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.1.17...@tds/shared-video-splash@1.1.18) (2019-10-04)

**Note:** Version bump only for package @tds/shared-video-splash





## [1.1.17](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.1.16...@tds/shared-video-splash@1.1.17) (2019-09-23)

**Note:** Version bump only for package @tds/shared-video-splash





## [1.1.16](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.1.15...@tds/shared-video-splash@1.1.16) (2019-09-12)

**Note:** Version bump only for package @tds/shared-video-splash





## [1.1.15](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.1.14...@tds/shared-video-splash@1.1.15) (2019-09-06)

**Note:** Version bump only for package @tds/shared-video-splash





## [1.1.14](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.1.13...@tds/shared-video-splash@1.1.14) (2019-08-26)

**Note:** Version bump only for package @tds/shared-video-splash





## [1.1.13](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.1.12...@tds/shared-video-splash@1.1.13) (2019-08-23)

**Note:** Version bump only for package @tds/shared-video-splash





## [1.1.12](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.1.11...@tds/shared-video-splash@1.1.12) (2019-08-22)

**Note:** Version bump only for package @tds/shared-video-splash





## [1.1.11](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.1.10...@tds/shared-video-splash@1.1.11) (2019-08-09)

**Note:** Version bump only for package @tds/shared-video-splash





## [1.1.10](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.1.9...@tds/shared-video-splash@1.1.10) (2019-08-01)

**Note:** Version bump only for package @tds/shared-video-splash





## [1.1.9](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.1.8...@tds/shared-video-splash@1.1.9) (2019-07-31)

**Note:** Version bump only for package @tds/shared-video-splash





## [1.1.8](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.1.7...@tds/shared-video-splash@1.1.8) (2019-07-30)

**Note:** Version bump only for package @tds/shared-video-splash





## [1.1.7](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.1.6...@tds/shared-video-splash@1.1.7) (2019-07-25)

**Note:** Version bump only for package @tds/shared-video-splash





## [1.1.6](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.1.5...@tds/shared-video-splash@1.1.6) (2019-07-17)

**Note:** Version bump only for package @tds/shared-video-splash





## [1.1.5](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.1.4...@tds/shared-video-splash@1.1.5) (2019-07-11)

**Note:** Version bump only for package @tds/shared-video-splash





## [1.1.4](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.1.3...@tds/shared-video-splash@1.1.4) (2019-07-10)

**Note:** Version bump only for package @tds/shared-video-splash





## [1.1.3](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.1.2...@tds/shared-video-splash@1.1.3) (2019-07-10)

**Note:** Version bump only for package @tds/shared-video-splash





## [1.1.2](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.1.1...@tds/shared-video-splash@1.1.2) (2019-07-02)

**Note:** Version bump only for package @tds/shared-video-splash





## [1.1.1](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.1.0...@tds/shared-video-splash@1.1.1) (2019-07-02)


### Bug Fixes

* **shared-video-splash:** correct big play button alignment on IE 11 ([7a13ab6](https://github.com/telus/tds-core/commit/7a13ab6))





# [1.1.0](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.0.6...@tds/shared-video-splash@1.1.0) (2019-06-17)


### Features

* **shared-video-splash:** add support for custom start button ([096367d](https://github.com/telus/tds-core/commit/096367d))





## [1.0.6](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.0.5...@tds/shared-video-splash@1.0.6) (2019-06-04)

**Note:** Version bump only for package @tds/shared-video-splash

## [1.0.5](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.0.4...@tds/shared-video-splash@1.0.5) (2019-05-01)

**Note:** Version bump only for package @tds/shared-video-splash

## [1.0.4](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.0.3...@tds/shared-video-splash@1.0.4) (2019-04-23)

**Note:** Version bump only for package @tds/shared-video-splash

## [1.0.3](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.0.2...@tds/shared-video-splash@1.0.3) (2019-04-22)

**Note:** Version bump only for package @tds/shared-video-splash

## [1.0.2](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.0.1...@tds/shared-video-splash@1.0.2) (2019-04-16)

**Note:** Version bump only for package @tds/shared-video-splash

## [1.0.1](https://github.com/telus/tds-core/compare/@tds/shared-video-splash@1.0.0...@tds/shared-video-splash@1.0.1) (2019-04-11)

**Note:** Version bump only for package @tds/shared-video-splash

# 1.0.0 (2019-04-01)

### Features

- **shared-video-splash:** add shared video splash component ([0308fa3](https://github.com/telus/tds-core/commit/0308fa3))
