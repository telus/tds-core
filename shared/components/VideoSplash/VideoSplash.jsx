import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import BigVideoButton from './BigVideoButton/BigVideoButton'

import Play from './svg/Play'

import safeRest from '../../utils/safeRest'

/**
 * @version ./package.json
 */

const SplashBackground = styled.div(props => ({
  backgroundImage: `url(${props.poster})`,
  backgroundSize: 'cover',
  backgroundPosition: 'center',
  position: 'absolute',
  top: 0,
  width: '100%',
  height: '100%',
  cursor: 'pointer',
}))

const VideoSplash = ({ poster, videoLength, label, ...rest }) => {
  return (
    <SplashBackground {...safeRest(rest)} poster={poster}>
      <BigVideoButton icon={<Play />} label={label} videoLength={videoLength} />
    </SplashBackground>
  )
}

VideoSplash.propTypes = {
  poster: PropTypes.string,
  videoLength: PropTypes.number.isRequired,
  label: PropTypes.string.isRequired,
}

VideoSplash.defaultProps = {
  poster: undefined,
}

export default VideoSplash
