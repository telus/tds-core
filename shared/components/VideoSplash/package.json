{
  "name": "@tds/shared-video-splash",
  "version": "1.0.0-dev",
  "private": true,
  "main": "",
  "module": "",
  "dependencies": {
    "@tds/core-box": "^1.1.0",
    "@tds/core-decorative-icon": "^1.1.1",
    "@tds/core-paragraph": "^1.0.6",
    "@tds/core-responsive": "^1.2.0",
    "prop-types": "^15.5.10"
  }
}
