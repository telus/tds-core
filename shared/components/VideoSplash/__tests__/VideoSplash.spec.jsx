import React from 'react'
import { shallow, mount } from 'enzyme'

import VideoSplash from '../VideoSplash'

describe('VideoSplash', () => {
  const doShallow = (props = {}) =>
    shallow(<VideoSplash label="Watch the video" poster="" videoLength={61} {...props} />)

  const doMount = (props = {}) =>
    mount(<VideoSplash label="Watch the video" poster="" videoLength={61} {...props} />)

  it('renders', () => {
    const videoSplash = doShallow()

    expect(videoSplash).toMatchSnapshot()
  })

  it('displays timestamps correctly', () => {
    const videoSplash = doMount()
      .find('[data-testid="timestamp"]')
      .find('p')

    expect(videoSplash.text()).toEqual('1:01')
  })

  it('passes additional attributes to the element', () => {
    const videoSplash = doShallow({ id: 'the-id', 'data-some-attr': 'some value' })

    expect(videoSplash).toHaveProp('id', 'the-id')
    expect(videoSplash).toHaveProp('data-some-attr', 'some value')
  })

  it('does not allow custom CSS', () => {
    const videoSplash = doShallow({
      className: 'my-custom-class',
      style: { color: 'hotpink' },
    })

    expect(videoSplash).not.toHaveProp('className', 'my-custom-class')
    expect(videoSplash).not.toHaveProp('style')
  })
})
