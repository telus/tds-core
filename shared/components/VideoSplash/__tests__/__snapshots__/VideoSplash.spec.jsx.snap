// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoSplash renders 1`] = `
<VideoSplash__SplashBackground
  poster=""
>
  <BigVideoButton
    icon={<Play />}
    label="Watch the video"
    videoLength={61}
  />
</VideoSplash__SplashBackground>
`;
